specs/Print.md

```diff
@@ -0,0 +1,425 @@
+"Ctrl + Shift + P" or clicking "Print using system dialog" link in print preview dialog. This API gives developers the ability to open print dialog without requiring the end user interaction and also can print without a dialog to a specific printer by configuring the print settings. 
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose following API's 
+
+**Print API**: This API gives developers the ability to open a browser print preview dialog or system print dialog without requiring the end user interaction. This API consists of a Print method and can either open a browser print preview dialog or system print dialog based on COREWEBVIEW2_PRINT_DIALOG_KIND value.
+
+**Print without a print dialog**: This API gives developers the ability to print the current page without a print dialog to the specific printer. This API consists of an asynchronous PrintWithSettings method and a PrintSettings2 object. 
+PrintSettings2 object extends PrintSettings to support printing to a printer.
+
+**Print to PDF Stream**: This API gives developers printable PDF data, which can be used as a preview in a custom print preview dialog. This API consists of an asynchronous PrintToPdfStream method and a PrintSettings2 object. 
+
+
+# Examples
+## Print
+
+You can use `Print` method to programmatically open a print dialog without end user interaction. Currently devs are using Javascript window.print() using WebView2 ExecuteScriptAsync api to invoke
+browser print preview dialog and there isn't a way to invoke open system print dialog programmatically.
+
+// This example shows the user a print dialog. If `printDialogKind` is COREWEBVIEW2_PRINT_DIALOG_KIND_BROWSER_PRINT_PREVIEW, opens a browser print preview dialog, otherwise opens a system print dialog.       
+```cpp
+bool AppWindow::OpenPrintDialog(COREWEBVIEW2_PRINT_DIALOG_KIND printDialogKind)
+{
+     wil::com_ptr<ICoreWebView2_15> webView2_15;
+     CHECK_FAILURE(m_webView->QueryInterface(IID_PPV_ARGS(&webView2_15)));
+     CHECK_FEATURE_RETURN(webView2_15);
+     CHECK_FAILURE(webView2_15->Print(printDialogKind));
+     return true;
+}
+```
+
+```c#
+void OpenPrintDialog(object target, ExecutedRoutedEventArgs e)
+{
+     string printDialog = e.Parameter.ToString();
+     if (printDialog == "PrintPreview")
+     {
+         webView.CoreWebView2.Print(CoreWebView2PrintDialogKind.BrowserPrintPreview);
+     }
+     else
+     {
+         webView.CoreWebView2.Print(CoreWebView2PrintDialogKind.SystemPrint);
+     }
+}
+```
+
+## PrintWithSettings
+
+You can use `PrintWithSettings` method to print to a specific printer without a print dialog. You can programatically configure print settings using `CreatePrintSettings2`, and call `PrintWithSettings` to print the current page.
+
+// This example prints the current page without a print dialog to the `Microsoft Print To PDF` printer with the settings.
```

Comment:
Is there a reason why a dev would do all this to print to a pdf versus using our existing CoreWebView2.PrintToPdf method? Maybe we can use a different printer name or take the name as a parameter or something?